src/web3/saga.ts

```diff
@@ -154,25 +154,26 @@ function* getAddress<T extends { address: string | null }>({
   action: Actions
 }) {
   while (true) {
-    const account = yield select(addressSelector)
+    const account = yield* select(addressSelector)
     if (account) {
       return account
     }
 
-    const actionEffect: T = yield take(action)
-    if (actionEffect.address) {
+    const actionEffect = yield* take(action)
```

Comment:
nit: can we do something like this instead?
```suggestion
    const actionEffect = (yield* take(action) as T)
```

Reply:
I can move it up there, but we still have to cast as unknown first